### PR TITLE
Application approved

### PR DESCRIPTION
**Pull request by janegreene**
Completed user stories 26-7, 31, 34-6
Made adjustments to application approve and revoke

### CODE REVIEW

File: app/controllers/apps_controller.rb [hunk 1]
```diff
@@ -31,12 +31,64 @@ def show
...
+      pet.update(status: "Pending")
+      pet_app = pet.pet_applications.find_by(application_id: params[:id])
+      pet_app.update(approved: true)
+      redirect_to "/pets/#{pet.id}"
+    end
+  end
+
+  def unapprove
+    @app = Application.find(params[:id])
+    pet_app = PetApplication.where(application_id: @app.id).first #need to change .first if we do single(or multi)
```
Comment (janegreene) [inline app/controllers/apps_controller.rb:77]: this needs revision

File: app/views/apps/show.html.erb [hunk 2]
```diff
@@ -7,10 +7,17 @@
...
-<%= link_to "Approve Application", "/pets/#{pet.id}/change_status", method: :patch %>
+
+<h1>Approve Application</h1>
+
+<% @app.pets.each do |pet| %>
+      <% if pet.on_hold_for == nil %>
+        <%= button_to "Approve Application for #{pet.name}", "/applications/#{@app.id}/update", method: :patch, params: {pet_id: pet.id} %>
+      <% else %>
+       <% if pet.on_hold_for == @app.name %>
+        <%= button_to "Unapprove Application for #{pet.name}", "/applications/#{@app.id}/unapprove", method: :patch %>
```
Comment (whitneykidd) [inline app/views/apps/show.html.erb:18]: This looks a lot better than what we previously had ⭐

File: app/models/pet.rb [hunk 3]
```diff
@@ -1,7 +1,14 @@
 class Pet < ApplicationRecord
-  validates_presence_of :name
+  validates_presence_of :name, :image, :sex, :age, :description
   belongs_to :shelter
-  has_many :pet_applications
+  has_many :pet_applications, dependent: :destroy
   has_many :applications, through: :pet_applications
 
+  def on_hold_for
```
Comment (janegreene) [inline app/models/pet.rb:7]: This is handy! Whitney will fix html later.

File: app/views/pets/show.html.erb [hunk 4]
```diff
@@ -1,10 +1,6 @@
 <div class="container">
+<b>On hold for: <%= @pet.on_hold_for%> </b><br/>
```
Comment (whitneykidd) [inline app/views/pets/show.html.erb:2]: I'll format this in the future(tomorrow)

File: config/routes.rb [hunk 5]
```diff
@@ -30,7 +30,9 @@
   get '/applications/new', to: 'apps#new'
   get '/applications/:id', to: 'apps#show'
   post '/applications/new', to: 'apps#create'
-  patch '/pets/:id/change_status', to: 'pets#change_status'
+  patch 'applications/:id/update', to: 'apps#update'
+  patch 'applications/:id/unapprove', to: 'apps#unapprove'
+  # patch '/pets/:id/change_status', to: 'pets#change_status'
```
Comment (janegreene) [inline config/routes.rb:35]: will remove before final submit

File: spec/features/applications/show_spec.rb [hunk 6]
```diff
@@ -47,17 +47,57 @@
...
-end
 
+  it "Pets can only have one approved application on them at a time" do
+    application2 = Application.create(name: "Bill Rogers",
+      address: "132 Maple Dr.", city: "Claremore", state: "OK", zip: 74014, phone: "918-233-9000",
+      description: "great fenced yard", pet_ids: ["#{@pet1.id}", "#{@pet2.id}"])
+
+    visit "/applications/#{@application.id}"
+
+    click_button "Approve Application for Milo"
```
Comment (janegreene) [inline spec/features/applications/show_spec.rb:75]: replaced the select

**Pull request merged**